cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/BatchSparkPipelineDriver.java

```diff
@@ -198,12 +198,14 @@ public void run(DatasetContext context) throws Exception {
         sec.getRuntimeArguments().getOrDefault(Constants.CONSOLIDATE_STAGES, Boolean.TRUE.toString()));
       boolean shouldCacheFunctions = Boolean.parseBoolean(
         sec.getRuntimeArguments().getOrDefault(Constants.CACHE_FUNCTIONS, Boolean.TRUE.toString()));
+      boolean shouldDisablePushdown = Boolean.parseBoolean(
+        sec.getRuntimeArguments().getOrDefault(Constants.DISABLE_ELT_PUSHDOWN, Boolean.FALSE.toString()));
       boolean isPreviewEnabled =
         phaseSpec.getPhase().size() == 0
           || sec.getDataTracer(phaseSpec.getPhase().iterator().next().getName()).isEnabled();
 
       // Initialize SQL engine instance if needed.
-      if (!isPreviewEnabled && phaseSpec.getSQLEngineStageSpec() != null) {
+      if (!isPreviewEnabled && phaseSpec.getSQLEngineStageSpec() != null && !shouldDisablePushdown) {
```

Comment:
do we need to add a test for this?

Reply:
I'll add a unit tests. Next week you should add an integration tests that covers this flag as well.